cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionConsumerResult.java

```diff
@@ -24,9 +24,10 @@
  */
 public class PartitionConsumerResult {
   private final PartitionConsumerState partitionConsumerState;
-  private final Iterator<Partition> partitionIterator;
+  private final List<PartitionDetail> partitionIterator;
```

Comment:
renamed.
Partitions that expire go back into the `NEW` state of the working set (same behavior as partitions that get aborted/fail processing).